zerver/views/message_send.py

```diff
@@ -123,21 +128,30 @@ def same_realm_jabber_user(user_profile: UserProfile, email: str) -> bool:
     return RealmDomain.objects.filter(realm=user_profile.realm, domain=domain).exists()
 
 
-@has_request_variables
+@typed_endpoint
 def send_message_backend(
     request: HttpRequest,
     user_profile: UserProfile,
-    req_type: str = REQ("type", str_validator=check_string_in(Message.API_RECIPIENT_TYPES)),
-    req_to: str | None = REQ("to", default=None),
-    req_sender: str | None = REQ("sender", default=None, documentation_pending=True),
-    forged_str: str | None = REQ("forged", default=None, documentation_pending=True),
-    topic_name: str | None = REQ_topic(),
-    message_content: str = REQ("content"),
-    widget_content: str | None = REQ(default=None, documentation_pending=True),
-    local_id: str | None = REQ(default=None),
-    queue_id: str | None = REQ(default=None),
-    time: float | None = REQ(default=None, converter=to_float, documentation_pending=True),
-    read_by_sender: bool | None = REQ(json_validator=check_bool, default=None),
+    *,
+    req_type: Annotated[Literal["direct", "private", "stream", "channel"], ApiParamConfig("type")],
+    to: Json[str | int | list[int] | list[str]] | None = None,
```

Comment:
We could accept raw strings and do the parsing later, like we did previously
```
to: str = None
```
But this causes the OpenApi tests to complain about it not being `Json`.